### ISSUE TITLE
Make ``modified-iterating-*`` also check class attributes

### ISSUE THREAD

**Issue by DanielNoord** [PyCQA/pylint#7391]
### Current problem

See #7386 for some discussion.

Basically:
```python
class MyClass:
    def __init__(self) -> None:
        self.attribute = [1, 2, 3]

   def my_other_method(self):
        """This should raise as we are appending."""
        for var in self.attribute:
            self.attribute.append(var)  # [modified-iterating-list]
```

Doesn't raise currently and it should. This likely requires some refactoring of the current checker as there are some assumptions in the checks we use there that aren't true with respect to class attributes.

### Desired solution

Emit the warning as described.

Tagging @orSolocate so I can assign them.
